src/hotspot/os/posix/gc/z/zVirtualMemory_posix.cpp

```diff
@@ -33,53 +33,22 @@ void ZVirtualMemoryManager::initialize_os() {
   // Does nothing
 }
 
-static void unmap(uintptr_t start, size_t size) {
-  const int res = munmap((void*)start, size);
-  assert(res == 0, "Failed to unmap memory");
-}
-
-static bool map(uintptr_t start, size_t size) {
-  const void* const res = mmap((void*)start, size, PROT_NONE, MAP_ANONYMOUS|MAP_PRIVATE|MAP_NORESERVE, -1, 0);
-  if (res == MAP_FAILED) {
+uintptr_t ZVirtualMemoryManager::os_reserve(uintptr_t addr, size_t size) {
+  const uintptr_t res = (uintptr_t)mmap((void*)addr, size, PROT_NONE, MAP_ANONYMOUS|MAP_PRIVATE|MAP_NORESERVE, -1, 0);
+  if (res == (uintptr_t)MAP_FAILED) {
     // Failed to reserve memory
-    return false;
+    return 0;
   }
 
-  if ((uintptr_t)res != start) {
+  if (res != addr) {
     // Failed to reserve memory at the requested address
-    unmap((uintptr_t)res, size);
-    return false;
+    os_unreserve(res, size);
```

Comment:
A bit odd to see the assymetry here: `mmap` is used above, so `munmap` would be expected here? I see `os_unreserve` does the same thing, but does it guaranteed to do so?

Reply:
Yes. I agree. Will fix.